server/src/tenant/metadata/migration-generator/migration-generator.service.ts

```diff
@@ -1,72 +1,44 @@
 import { Injectable } from '@nestjs/common';
 
-import { IsNull, QueryRunner, Table, TableColumn } from 'typeorm';
+import { QueryRunner, Table, TableColumn } from 'typeorm';
 
 import { DataSourceService } from 'src/tenant/metadata/data-source/data-source.service';
-
 import {
-  Migration,
-  MigrationColumn,
-  TenantMigration,
-} from './tenant-migration.entity';
+  TenantMigrationTableChange,
+  TenantMigrationColumnChange,
+} from 'src/tenant/metadata/tenant-migration/tenant-migration.entity';
+import { TenantMigrationService } from 'src/tenant/metadata/tenant-migration/tenant-migration.service';
 
 @Injectable()
 export class MigrationGeneratorService {
-  constructor(private readonly dataSourceService: DataSourceService) {}
-
-  private async getPendingMigrations(workspaceId: string) {
```

Comment:
See below in this PR, I moved all of this to a dedicated tenant-migration service to separate the TenantMigration repository logic and the migration generation coming from it